src/storage/src/source/mysql/replication.rs

```diff
@@ -531,7 +537,11 @@ async fn raw_stream<'a>(
          and server_id: {server_id:?}"
     );
 
-    let repl_stream = match conn
+    // We need to transform the connection into a BinlogStream, but to avoid dropping any active
+    // SSH tunnel used by the connection we need to preserve the tunnel handle and return it
+    let (inner_conn, conn_tunnel_handle) = conn.take();
```

Comment:
```suggestion
    // We need to transform the connection into a BinlogStream (it takes the `Conn` by value), but to avoid dropping any active
    // SSH tunnel used by the connection we need to preserve the tunnel handle and return it
    let (inner_conn, conn_tunnel_handle) = conn.take();
```